zkevm-circuits/src/bytecode_circuit/bytecode_unroller.rs

```diff
@@ -0,0 +1,865 @@
+use crate::{
+    evm_circuit::op_execution::utils::{
+        and, constraint_builder::ConstraintBuilder, from_bytes, not, or, select,
+    },
+    gadget::is_zero::{IsZeroChip, IsZeroConfig, IsZeroInstruction},
+    util::Expr,
+};
+use bus_mapping::evm::OpcodeId;
+use halo2::{
+    circuit::{Layouter, Region},
+    plonk::{
+        Advice, Column, ConstraintSystem, Error, Expression, Fixed, Selector,
+    },
+    poly::Rotation,
+};
+use keccak256::plain::Keccak;
+use pasta_curves::arithmetic::FieldExt;
+use std::marker::PhantomData;
+
+use super::param::{
+    BLOCK_SIZE, KECCAK_INPUT_WIDTH, KECCAK_OUTPUT_WIDTH, WORD_SIZE,
+};
+
+#[derive(Clone, Debug)]
+pub struct BytecodeUnrollerConfig<F> {
```

Comment:
I think placing  bytecode circuit to bussmapping repo more reasonable, like  rw_table , evm _circuit just take use of it, maybe good modularization, what do you think?

Reply:
I followed the example of the state circuit which is also put into a subfolder in zkevm-circuits. For the actual circuit part I think you would expect the repo named `zkevm-circuits` to, well, contain the circuits for zkevm. :) But I think you may be right that the busmapping repo may be the place to generate some of the input data for the circuit.